plugins/LilyPad/LilyPad.cpp

```diff
@@ -163,7 +163,7 @@ struct Stick {
 
 // Sum of states of all controls for a pad (Not including toggles).
 struct ButtonSum {
-	int buttons[12];
+	int buttons[16];
 	Stick sticks[3];
```

Comment:
You'll need to change the save state version.

Personally I wonder whether it should be
```
struct ButtonSum {
	int buttons[16];
	Stick sticks[2];
};
```
so that we only treat the d-pad as buttons. I haven't looked too deeply into the idea though.